sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -203,6 +203,260 @@ In the case of `ext_intel_packed` matrix memory layout, `row` and
+          layout Layout, access::address_space Space,
+          access::decorated IsDecorated>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, use::a, Rows, Cols, Layout> &res,
+   multi_ptr<T, Space, IsDecorated> base_dest, size_t Stride,
+   size_t GlobalRows, size_t GlobalCols, size_t RowIndex, size_t ColIndex);
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          layout Layout, access::address_space Space,
+          access::decorated IsDecorated>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, use::b, Rows, Cols, Layout> &res,
+   multi_ptr<T, Space, IsDecorated> base_dest, size_t Stride,
+   size_t GlobalRows, size_t GlobalCols, size_t RowIndex, size_t ColIndex);
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          typename PropertyListT>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, use::accumulator, Rows, Cols, layout::dynamic> &res, 
+   ext::oneapi::experimental::annotated_ptr<T, PropertyListT> base_dest,
+   size_t Stride, layout Layout, size_t GlobalRows, size_t GlobalCols,
+   size_t RowIndex, size_t ColIndex);
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          layout Layout, typename PropertyListT>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, use::a, Rows, Cols, Layout> &res,
+   ext::oneapi::experimental::annotated_ptr<T, PropertyListT> base_dest,
+   size_t Stride, size_t GlobalRows, size_t GlobalCols,
+   size_t RowIndex, size_t ColIndex);
+
+template <typename Group, typename T, size_t Rows, size_t Cols,
+          layout Layout, typename PropertyListT>
+void joint_matrix_store_checked(Group g,
+   const joint_matrix<Group, T, use::b, Rows, Cols, Layout> &res,
+   ext::oneapi::experimental::annotated_ptr<T, PropertyListT> base_dest,
+   size_t Stride, size_t GlobalRows, size_t GlobalCols,
+   size_t RowIndex, size_t ColIndex);
+
+} // namespace sycl::ext::intel::experimental::matrix
+```
+
+The property list associated with the `annotated_ptr` argument
+represents the compile-time constant properties for cache control included
+in the SYCL extenion
+link:../../proposed/sycl_ext_intel_cache_controls.asciidoc[sycl_ext_intel_cache_controls].
+
+==== Restrictions and Device Information Descriptors
+Applications must adhere to certain alignment restrictions when using
+the checked APIs described in this section. This extension provides
+the following queries to get these requirements:
+
+[frame="none",options="header"]
+|======================
+| Device descriptors | Return type| Description
+|`ext::intel::experimental::info::device::matrix_checked_alignment`| `size_t`
+|Tells the required alignment (in bytes) of the base pointer for
+`joint_matrix_load_checked` and `joint_matrix_store_checked`.
+|`ext::intel::experimental::info::device::matrix_checked_rowindex_multiple_of<T>`|
+`size_t`|Returns a value, of which `RowIndex` must be multiple of.
```

Comment:
Kind of a nit, but it would be good to add a sentence:

> Where `T` is the element type of the matrix.

However, this makes me wonder how this works for a matrix that uses TF32.  Should `T` be `precision::tf32` or `float` (which is the storage type when using TF32)?

Reply:
We should be able to use the element type `precision::tf32`

Reply:
I'd suggest adding another sentence to make this clear.  Something like:

> When using the matrices with the machine learning types, `T` should be the element type (e.g. `precision::tf32`) not the storage type.